src/storage/mvcc/txn.rs

```diff
@@ -149,14 +149,18 @@ impl<S: Snapshot> MvccTxn<S> {
         value: Option<Value>,
         options: &Options,
     ) {
-        if value.is_none() || is_short_value(value.as_ref().unwrap()) {
-            // If the value is short, embed it in Lock.
-            self.lock_key(key, lock_type, primary, value, options);
-        } else {
-            // value is long
-            let ts = self.start_ts;
-            self.put_value(key.clone(), ts, value.unwrap());
+        if let Some(value) = value {
```

Comment:
Does the previous implementation make `unnecessary_unwrap` lint unhappy?

Reply:
Yep. It figures out the unwrap check is redundant with the check in the `if` branch.

I think this example is a pretty good application of the lint, even though the code is more verbose, since it takes a bit of thinking to figure out it's ok to unwrap both here and in the `if` header. The transformed version is correct by construction.

Reply:
Em, I don't like the style which clippy suggests here 🤔